lib/ecto/changeset.ex

```diff
@@ -28,16 +30,16 @@ defmodule Ecto.Changeset do
             filters: %{}
 
   @type error :: {atom, atom | {atom, [term]}}
-  @type t :: %Ecto.Changeset{valid?: boolean(),
-                             repo: atom | nil,
-                             model: Ecto.Model.t | nil,
-                             params: %{String.t => term} | nil,
-                             changes: %{atom => term},
-                             required: [atom],
-                             optional: [atom],
-                             errors: [error],
-                             validations: [{atom, atom | {atom, [term]}}],
-                             filters: %{atom => term}}
+  @type t :: %__MODULE__{valid?: boolean(),
```

Comment:
I looked at `Plug.Conn` and it does the same thing there, `__MODULE__` in the struct but `%Conn{}` in the functions. Should I change this?